tests/integration/telemetry/stats/prometheus/TCP/stats_tcp_filter_test.go

```diff
@@ -0,0 +1,193 @@
+	"istio.io/istio/pkg/test/framework/components/bookinfo"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/ingress"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/mixer"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	util "istio.io/istio/tests/integration/mixer"
+	util_prometheus "istio.io/istio/tests/integration/telemetry/stats/prometheus"
+)
+
+const (
+	metadataExchangeFilterConfig = "testdata/metadata_exchange_tcp_filter.yaml"
+	statsFilterConfig            = "testdata/stats_filter.yaml"
+	cleanupFilterConfig          = "testdata/cleanup.yaml"
+)
+
+var (
+	ist        istio.Instance
+	bookinfoNs namespace.Instance
+	g          galley.Instance
+	ing        ingress.Instance
+	prom       prometheus.Instance
+)
+
+func TestTcpMetric(t *testing.T) { // nolint:interfacer
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			addr := ing.HTTPAddress()
+			url := fmt.Sprintf("http://%s/productpage", addr.String())
+			g.ApplyConfigOrFail(
+				t,
+				bookinfoNs,
+				bookinfo.GetDestinationRuleConfigFileOrFail(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+				bookinfo.NetworkingTCPDbRule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+			)
+			defer g.DeleteConfig(
+				bookinfoNs,
+				bookinfo.GetDestinationRuleConfigFileOrFail(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+				bookinfo.NetworkingTCPDbRule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+			)
+
+			systemNM := namespace.ClaimSystemNamespaceOrFail(ctx, ctx)
+			mxc, err := file.AsString(metadataExchangeFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", metadataExchangeFilterConfig, err)
+			}
+			stats, err := file.AsString(statsFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", statsFilterConfig, err)
+			}
+
+			cleanup, err := file.AsString(cleanupFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", metadataExchangeFilterConfig, err)
+			}
+
+			g.ApplyConfigOrFail(
+				t,
+				systemNM,
+				mxc,
+				stats,
+				cleanup,
+			)
+			defer g.DeleteConfig(
+				systemNM,
+				mxc,
+				stats,
+				cleanup,
+			)
+
+			util.AllowRuleSync(t)
+
+			destinationQuery := buildQuery()
+			retry.UntilSuccessOrFail(t, func() error {
+				util.SendTraffic(ing, t, "Sending traffic", url, "", 200)
+				if err := util_prometheus.QueryPrometheus(t, destinationQuery, prom); err != nil {
+					t.Logf("prometheus values for istio_requests_total: \n%s", util.PromDump(prom, "istio_requests_total"))
```

Comment:
`istio_requests_total` is a http metric. I think a TCP metric is supposed to be tested here?

Reply:
istio_requests_total if for both http and tcp now. It has protocol field that differentiates between the two.

Reply:
Why aren't we following what we have right now? https://istio.io/docs/reference/config/policy-and-telemetry/metrics/ Our dashboard built on the assumption about the categorization of the metrics. And I think there would be user's customized dashboard based on this assumption too. I looked into proxy's code and looks like we are actually not reporting those four tcp metrics?

Reply:
We added TCP as protocol in existing metrics.. I can make a change to proxy first and then here to match what we had for 1.4..

Reply:
Thanks.. Added a TODO for the same.